homeassistant/components/roomba/__init__.py

```diff
@@ -1 +1,129 @@
 """The roomba component."""
+import asyncio
+import logging
+
+import async_timeout
+from roomba import Roomba, RoombaConnectionError
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+
+from .const import COMPONENTS, CONF_CERT, CONF_CONTINUOUS, CONF_DELAY, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass, config):
+    """Set up the roomba environment."""
+    if DOMAIN not in config:
+        return True
+
+    if not hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data={}
+            )
+        )
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set the config entry up."""
+    # Set up roomba platforms with config entry
+    if config_entry.data is None:
+        return False
+
+    if not config_entry.options:
+        hass.config_entries.async_update_entry(
+            config_entry,
+            options={
+                "certificate": config_entry.data[CONF_CERT],
```

Comment:
```suggestion
```
If I understand this correctly, the certificate is needed for authentication?  If so it should part of the config flow instead of the options flow.